### ISSUE TITLE
Analyzer creates extremely large result for GoMod project and creates issues during scanning phase

### ISSUE THREAD

**Issue by nate-droid** [oss-review-toolkit/ort#3860]
I am currently running an Analyze on this [repo](https://github.com/valocode/bubbly/). In addition to taking over 10 minutes, the result is approx 10GB in size. The entire project is about 300mb, so this seems a bit too large. 

When running Scan on the results, it will take between 30-40 min before it will eventually result in `java.lang.OutOfMemoryError: Java heap space`. This process has been confirmed on several machines.

`ort analyze -i ../path-my-project -o ../path-to-output -m GoMod`
`ort scan -i ../output/analyzer-result.yml -o ../output --skip-excluded`

**Comment by sschuberth** [comment]
Probably one cause is that we haven't ported the Go-analyzers to the new dependency graph format yet, also see https://github.com/oss-review-toolkit/ort/issues/3825.

**Comment by sschuberth** [comment]
As a side note, please be aware that Go*Mod* support is still marked as *experimental* "as only proxy-based source artifacts but no VCS locations are supported".

**Comment by sschuberth** [comment]
While #4119 is already merged, #4249 still needs to be implemented to fully resolve this issue.

**Comment by fviernau** [comment]
I've recently did a larger re-work of ORT's GoMod support which affected the reported dependency tree. So, I re-tested this bug with:
```
./cli/build/install/ort/bin/ort --info analyze -i ~/sources/github/valocode/bubbly -o . -m gomod
``` 
This took less than two minutes to execute and produced a result of size 0.16MB without any analyzer issues.
The analyzer examined revision aca7f8deb2e02c62583a5a0e3599f5c1cb096a3b.

So, the issue is fixed and this ticket can be closed as fixed.
